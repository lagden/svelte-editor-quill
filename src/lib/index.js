// Reexport your entry components here
export { default as Editor } from './Editor.svelte'
