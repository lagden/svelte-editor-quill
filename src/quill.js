import Quill from 'quill'

export function quill(node, options) {
	const q = new Quill(node, {
		modules: {
			toolbar: [
				[{'header': 1}, {'header': 2}],
				['bold', 'italic', 'underline', 'strike'],
				['link', 'code-block']
			]
		},
		placeholder: 'Digite algo...',
		...options
	})

	const _container = node.querySelector('.ql-editor')

	const onTextChange = () => {
		const html = _container?.innerHTML ?? ''
		const customEvent = new CustomEvent('text-change', {
			detail: {
				html,
				text: q.getText()
			}
		})
		node.dispatchEvent(customEvent)
	}

	q.on('text-change', onTextChange)

	return {
		destroy() {
			q.off('text-change', onTextChange)
		}
	}
}
