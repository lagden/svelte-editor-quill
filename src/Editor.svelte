<script>
	import {quill} from './quill'

	export let options = {}
	export let data = ''

	let className = ''
	export {className as class}
	className = [...new Set(
		['_tadashi_svelte_editor_quill', ...className.split(' ')]
	)].join(' ')

	options = {
		theme: 'bubble',
		...options
	}
</script>

<div
	use:quill={options}
	on:text-change
	{...(className ? {class: className} : {})}
>{@html data}</div>

<style type="text/css">
	:root {
		--tadashi_svelte_editor_quill_border: 1px solid gray;
		--tadashi_svelte_editor_quill_height: auto;
	}

	._tadashi_svelte_editor_quill {
		border: var(--tadashi_svelte_editor_quill_border);
		height: var(--tadashi_svelte_editor_quill_height);
	}
</style>
