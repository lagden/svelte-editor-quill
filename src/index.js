/* istanbul ignore file */
import {quill} from './quill'
import Editor from './Editor.svelte'

export {
	Editor,
	quill
}
