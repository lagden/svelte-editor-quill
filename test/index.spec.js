import {cleanup, render} from '@testing-library/svelte'
import {Editor} from '../src'

describe('Editor Quill', () => {
	afterEach(cleanup)

	test('should match snapshot', () => {
		const {container} = render(Editor)
		expect(container).toMatchSnapshot()
	})
})
