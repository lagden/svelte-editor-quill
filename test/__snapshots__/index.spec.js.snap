// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor Quill should match snapshot 1`] = `
<body>
  <div>
    <div
      class="_tadashi_svelte_editor_quill svelte-6djrvn ql-container ql-bubble"
    >
      <div
        class="ql-editor ql-blank"
        contenteditable="true"
        data-gramm="false"
        data-placeholder="Digite algo..."
      >
        <p>
          <br />
        </p>
      </div>
      <div
        class="ql-clipboard"
        contenteditable="true"
        tabindex="-1"
      />
      <div
        class="ql-tooltip ql-hidden"
      >
        <span
          class="ql-tooltip-arrow"
        />
        <div
          class="ql-tooltip-editor"
        >
          <input
            data-formula="e=mc^2"
            data-link="https://quilljs.com"
            data-video="Embed URL"
            type="text"
          />
          <a
            class="ql-close"
          />
        </div>
        <div
          class="ql-toolbar"
        >
          <span
            class="ql-formats"
          >
            <button
              class="ql-header"
              type="button"
              value="1"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <path
                  class="ql-fill"
                  d="M10,4V14a1,1,0,0,1-2,0V10H3v4a1,1,0,0,1-2,0V4A1,1,0,0,1,3,4V8H8V4a1,1,0,0,1,2,0Zm6.06787,9.209H14.98975V7.59863a.54085.54085,0,0,0-.605-.60547h-.62744a1.01119,1.01119,0,0,0-.748.29688L11.645,8.56641a.5435.5435,0,0,0-.022.8584l.28613.30762a.53861.53861,0,0,0,.84717.0332l.09912-.08789a1.2137,1.2137,0,0,0,.2417-.35254h.02246s-.01123.30859-.01123.60547V13.209H12.041a.54085.54085,0,0,0-.605.60547v.43945a.54085.54085,0,0,0,.605.60547h4.02686a.54085.54085,0,0,0,.605-.60547v-.43945A.54085.54085,0,0,0,16.06787,13.209Z"
                />
                 
              </svg>
            </button>
            <button
              class="ql-header"
              type="button"
              value="2"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <path
                  class="ql-fill"
                  d="M16.73975,13.81445v.43945a.54085.54085,0,0,1-.605.60547H11.855a.58392.58392,0,0,1-.64893-.60547V14.0127c0-2.90527,3.39941-3.42187,3.39941-4.55469a.77675.77675,0,0,0-.84717-.78125,1.17684,1.17684,0,0,0-.83594.38477c-.2749.26367-.561.374-.85791.13184l-.4292-.34082c-.30811-.24219-.38525-.51758-.1543-.81445a2.97155,2.97155,0,0,1,2.45361-1.17676,2.45393,2.45393,0,0,1,2.68408,2.40918c0,2.45312-3.1792,2.92676-3.27832,3.93848h2.79443A.54085.54085,0,0,1,16.73975,13.81445ZM9,3A.99974.99974,0,0,0,8,4V8H3V4A1,1,0,0,0,1,4V14a1,1,0,0,0,2,0V10H8v4a1,1,0,0,0,2,0V4A.99974.99974,0,0,0,9,3Z"
                />
                 
              </svg>
            </button>
          </span>
          <span
            class="ql-formats"
          >
            <button
              class="ql-bold"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <path
                  class="ql-stroke"
                  d="M5,4H9.5A2.5,2.5,0,0,1,12,6.5v0A2.5,2.5,0,0,1,9.5,9H5A0,0,0,0,1,5,9V4A0,0,0,0,1,5,4Z"
                />
                 
                <path
                  class="ql-stroke"
                  d="M5,9h5.5A2.5,2.5,0,0,1,13,11.5v0A2.5,2.5,0,0,1,10.5,14H5a0,0,0,0,1,0,0V9A0,0,0,0,1,5,9Z"
                />
                 
              </svg>
            </button>
            <button
              class="ql-italic"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="13"
                  y1="4"
                  y2="4"
                />
                 
                <line
                  class="ql-stroke"
                  x1="5"
                  x2="11"
                  y1="14"
                  y2="14"
                />
                 
                <line
                  class="ql-stroke"
                  x1="8"
                  x2="10"
                  y1="14"
                  y2="4"
                />
                 
              </svg>
            </button>
            <button
              class="ql-underline"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <path
                  class="ql-stroke"
                  d="M5,3V9a4.012,4.012,0,0,0,4,4H9a4.012,4.012,0,0,0,4-4V3"
                />
                 
                <rect
                  class="ql-fill"
                  height="1"
                  rx="0.5"
                  ry="0.5"
                  width="12"
                  x="3"
                  y="15"
                />
                 
              </svg>
            </button>
            <button
              class="ql-strike"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke ql-thin"
                  x1="15.5"
                  x2="2.5"
                  y1="8.5"
                  y2="9.5"
                />
                 
                <path
                  class="ql-fill"
                  d="M9.007,8C6.542,7.791,6,7.519,6,6.5,6,5.792,7.283,5,9,5c1.571,0,2.765.679,2.969,1.309a1,1,0,0,0,1.9-.617C13.356,4.106,11.354,3,9,3,6.2,3,4,4.538,4,6.5a3.2,3.2,0,0,0,.5,1.843Z"
                />
                 
                <path
                  class="ql-fill"
                  d="M8.984,10C11.457,10.208,12,10.479,12,11.5c0,0.708-1.283,1.5-3,1.5-1.571,0-2.765-.679-2.969-1.309a1,1,0,1,0-1.9.617C4.644,13.894,6.646,15,9,15c2.8,0,5-1.538,5-3.5a3.2,3.2,0,0,0-.5-1.843Z"
                />
                 
              </svg>
            </button>
          </span>
          <span
            class="ql-formats"
          >
            <button
              class="ql-link"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <line
                  class="ql-stroke"
                  x1="7"
                  x2="11"
                  y1="7"
                  y2="11"
                />
                 
                <path
                  class="ql-even ql-stroke"
                  d="M8.9,4.577a3.476,3.476,0,0,1,.36,4.679A3.476,3.476,0,0,1,4.577,8.9C3.185,7.5,2.035,6.4,4.217,4.217S7.5,3.185,8.9,4.577Z"
                />
                 
                <path
                  class="ql-even ql-stroke"
                  d="M13.423,9.1a3.476,3.476,0,0,0-4.679-.36,3.476,3.476,0,0,0,.36,4.679c1.392,1.392,2.5,2.542,4.679.36S14.815,10.5,13.423,9.1Z"
                />
                 
              </svg>
            </button>
            <button
              class="ql-code-block"
              type="button"
            >
              <svg
                viewBox="0 0 18 18"
              >
                 
                <polyline
                  class="ql-even ql-stroke"
                  points="5 7 3 9 5 11"
                />
                 
                <polyline
                  class="ql-even ql-stroke"
                  points="13 7 15 9 13 11"
                />
                 
                <line
                  class="ql-stroke"
                  x1="10"
                  x2="8"
                  y1="5"
                  y2="13"
                />
                 
              </svg>
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</body>
`;
